src/OSmOSE/Dataset.py

```diff
@@ -1,35 +1,30 @@
+from __future__ import annotations
```

Comment:
what does this import stand for ?

Reply:
It's just a [matter of compatibility](https://docs.astral.sh/ruff/rules/future-required-type-annotation/) with Python 3.10- versions for [PEP 604 type unions](https://peps.python.org/pep-0604/).